src/instance.rs

```diff
@@ -221,13 +221,25 @@ unsafe impl<T> AsPyPointer for Py2<'_, T> {
 ///
 /// The advantage of this over `&Py2` is that it avoids the need to have a pointer-to-pointer, as Py2
 /// is already a pointer to an `ffi::PyObject``.
+///
+/// Similarly, this type is `Copy` and `Clone`, like any pointer.
 #[repr(transparent)]
 pub(crate) struct Py2Borrowed<'a, 'py, T>(
     NonNull<ffi::PyObject>,
     PhantomData<&'a Py<T>>,
     Python<'py>,
 );
 
+impl<'py, T> Py2Borrowed<'_, 'py, T> {
+    pub(crate) fn clone_ref(self) -> Py2<'py, T> {
+        unsafe { ffi::Py_INCREF(self.as_ptr()) };
+        Py2(
+            self.py(),
+            ManuallyDrop::new(unsafe { Py::from_non_null(self.0) }),
+        )
+    }
+}
```

Comment:
`.clone_ref()` becomes necessary because implementing `Clone` and `Copy` means that the previous `.clone()` calls I'd been using to convert `Py2Borrowed` -> `Py2` no longer compile as they instead will now just clone `Py2Borrowed`. (Previously they worked because they would first deref to `Py2` before cloning.)

I think this API probably makes sense, it's also consistent in naming with `Py::clone_ref`.

Reply:
This is now just `to_owned` as per #3680